spec/index.html

```diff
@@ -503,282 +556,330 @@ <h2>Example</h2>
+        &lt;/emu-grammar>
+        is a shorthand for:
+      &lt;/p>
+      &lt;emu-grammar type="definition" example>
+        HexIntegerLiteral :: `0` `x` HexDigits
+      &lt;/emu-grammar>
+    </code></pre>
+
+    <h3>Result</h3>
+    <aside class="result">
+      <p>
+        <emu-grammar type="definition" collapsed example>
+          HexIntegerLiteral :: `0x` HexDigits
+        </emu-grammar>
+        is a shorthand for:
+      </p>
+      <emu-grammar type="definition" example>
+        HexIntegerLiteral :: `0` `x` HexDigits
+      </emu-grammar>
+    </aside>
   </emu-clause>
 
   <emu-clause id="emu-production">
     <h1>emu-production</h1>
-    <p>This is the top level element that contains all grammar productions. Each production MUST include at least one right-hand side (see <a href="#emu-rhs">emu-rhs</a>).</p>
-    <p>The production will be displayed as an "inline" element flowing with text unless it is the immediate child of an emu-clause-like element or if its containing emu-grammar element is an immediate child of an emu-clause-like element.</p>
-    <h2>Attributes</h2>
+    <p>This is the top level element that contains a grammar production associating a shared <dfn>LHS</dfn> (left-hand side) <emu-xref href="#emu-nt" title></emu-xref> non-terminal symbol with one or more <emu-xref href="#emu-rhs" title></emu-xref> elements containing arbitrary <dfn>RHS</dfn> (right-hand side) sequences of terminal and non-terminal symbols.</p>
+    <p>The production will be displayed as an inline element flowing with text unless it is the immediate child of an emu-clause-like element or if its containing emu-grammar element is an immediate child of an emu-clause-like element.</p>
+
     <h2>Attributes</h2>
-    <p><b>Name:</b> Required. Name of the production (i.e. the non-terminal on the LHS).</p>
-    <p><b>Params:</b> Parameters for this production. Multiple parameters separated by commas.</p>
-    <p><b>Type:</b> Type of production, either "lexical" or "regexp". Default is blank (normal).</p>
-    <p><b>oneof:</b> If present, production is a one-of production. See DecimalDigit example above.</p>
-    <p><b>collapsed:</b> If present, production is displayed in collapsed format with LHS and RHS on the same line.</p>
+    <p><b>name:</b> Required: Name of the production (i.e. the LHS). Should conform with the syntax of <a href="https://www.unicode.org/reports/tr31/#D1">Unicode Standard Annex #31, UAX31-D1 Default Identifier Syntax</a> and not include any underscore (U+005F LOW LINE “<b>_</b>”).</p>
+    <p><b>collapsed:</b> If present, the production is displayed in collapsed format with LHS and RHS on the same line.</p>
+    <p><b>oneof:</b> If present, the production is a one-of production. See the DecimalDigit example <emu-xref href="#grammar">above</emu-xref>.</p>
+    <!-- what does it even mean for a LHS to be optional??? -->
+    <p><b>optional:</b> If present, the LHS is marked as optional.</p>
```

Comment:
I think this attribute should just be removed, but maybe there's a case I'm just not getting for
```html
<emu-production>
  <emu-nt optional="">LHS</emu-nt> <emu-geq>:</emu-geq>
  <emu-rhs>…</emu-rhs>
  …
</emu-production>
```

Reply:
Yeah I have no idea what that's for.

Also, I'm not sure it makes sense to support writing `emu-production` (etc) manually; the happy path is using grammarkdown notation. Possibly these should not be in the documentation at all. (They'll still be implicitly supported because grammarkdown emits these nodes, but we don't need to tell people about them, I think.)